src/inference/Inference.jl

```diff
@@ -123,19 +123,26 @@ end
 ######################
 # Default Transition #
 ######################
+# Default
+# Extended in contrib/inference/abstractmcmc.jl
+getstats(t) = nothing
 
-struct Transition{T, F<:AbstractFloat, S<:Union{NamedTuple, Nothing}}
+Base.@kwdef struct Transition{T, F<:AbstractFloat, S<:Union{NamedTuple, Nothing}}
     θ     :: T
     lp    :: F # TODO: merge `lp` with `stat`
-    stat  :: S
+    stat  :: S = nothing
 end
 
-Transition(θ, lp) = Transition(θ, lp, nothing)
+function Transition(vi::AbstractVarInfo, t)
+    θ = tonamedtuple(vi)
+    lp = getlogp(vi)
+    return Transition(θ=θ, lp=lp, stat=getstats(t))
```

Comment:
Just make this
```suggestion
function Transition(vi::AbstractVarInfo, t=nothing; nt::NamedTuple=NamedTuple())
    θ = merge(tonamedtuple(vi), nt)
    lp = getlogp(vi)
    return Transition(θ, lp, getstats(t))
```
and everything is covered.